### PR TITLE
[MM-38107] Upgrade server deps to v6

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Hopefully I haven't made too many silly mistakes.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-38107

### CODE REVIEW

File: logger/logger.go [hunk 1]
```diff
@@ -18,36 +20,46 @@ type Settings struct {
...
 func New(logSettings *Settings) *mlog.Logger {
-	return mlog.NewLogger(&mlog.LoggerConfiguration{
-		EnableConsole: logSettings.EnableConsole,
-		ConsoleJson:   logSettings.ConsoleJson,
-		ConsoleLevel:  strings.ToLower(logSettings.ConsoleLevel),
-		EnableFile:    logSettings.EnableFile,
-		FileJson:      logSettings.FileJson,
-		FileLevel:     strings.ToLower(logSettings.FileLevel),
-		FileLocation:  logSettings.FileLocation,
+	logger, _ := mlog.NewLogger()
```
Comment (streamer45) [inline logger/logger.go:28]: All this isn't great but it works so I'd defer it to later time.

**Review by isacikgoz — APPROVED** [review]
Looks good 👍

File: go.mod [hunk 2]
```diff
@@ -1,26 +1,99 @@
...
 	github.com/sergi/go-diff v1.1.0 // indirect
 	github.com/spf13/cobra v1.1.3
 	github.com/stretchr/testify v1.7.0
 	github.com/valyala/fasthttp v1.7.1 // indirect
-	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
+	golang.org/x/crypto v0.0.0-20210616213533-5ff15b29337e
 	gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d // indirect
 )
 
+require (
```
Comment (agnivade) [inline go.mod:24]: Why a separate require block?

File: loadtest/store/memstore/random.go [hunk 3]
```diff
@@ -68,12 +68,12 @@ func (s *MemStore) RandomTeam(st store.SelectionType) (model.Team, error) {
...
 }
 
-func excludeChannelType(st store.SelectionType, channelType string) bool {
-	m := map[store.SelectionType]string{
-		store.SelectNotPublic:  model.CHANNEL_OPEN,
-		store.SelectNotPrivate: model.CHANNEL_PRIVATE,
-		store.SelectNotDirect:  model.CHANNEL_DIRECT,
-		store.SelectNotGroup:   model.CHANNEL_GROUP,
+func excludeChannelType(st store.SelectionType, channelType model.ChannelType) bool {
+	m := map[store.SelectionType]model.ChannelType{
```
Comment (agnivade) [inline loadtest/store/memstore/random.go:72]: typed constants :heart_eyes:

File: loadtest/control/actions.go [hunk 4]
```diff
@@ -692,7 +688,7 @@ func Reload(u user.User) UserActionResponse {
 	}
 
 	// Getting unread teams.
-	_, err = u.GetTeamsUnread("")
+	_, err = u.GetTeamsUnread("", false)
```
Comment (agnivade) [inline loadtest/control/actions.go:691]: Wish we had the time to test CRT paths as well :disappointed:

**Review by agnivade — APPROVED** [review]
Nice!

File: go.mod (hunk 2 above)
Comment (streamer45) [reply go.mod:24]: No idea, that was all `go mod tidy`. I'll have a look.

File: loadtest/control/actions.go (hunk 4 above)
Comment (streamer45) [reply loadtest/control/actions.go:691]: Yeah, along with other N recent features. Does reliable websockets ring a bell? :stuck_out_tongue_closed_eyes:  ([MM-38327](https://mattermost.atlassian.net/browse/MM-38327))

File: go.mod (hunk 2 above)
Comment (streamer45) [reply go.mod:24]: Seems expected as of version 1.17:

https://tip.golang.org/doc/go1.17#go-command

**Pull request merged**